### PR TITLE
[RFC] doc: Remove more references to MS-DOS

### PR DESCRIPTION
**Pull request by sethjackson**
Just more MS-DOS references sent to the bit bucket.

References to MS-DOS as they refer to the file format `:set fileformat=dos` have been left as I feel that makes the most logical sense.

### CODE REVIEW

File: runtime/doc/cmdline.txt [hunk 1]
```diff
@@ -820,7 +820,7 @@ These modifiers can be given, in this order:
 		separator is removed.  Thus ":p:h" on a directory name results
 		on the directory name itself (without trailing slash).
 		When the file name is an absolute path (starts with "/" for
-		Unix; "x:\" for MS-DOS and WIN32), that part is not removed.
+		Unix; "x:\" for Windows), that part is not removed.
```
Comment (Pyrohh) [inline runtime/doc/cmdline.txt]: :+1: I'm glad to see that we're referring to Windows platforms in a more unified way.

File: runtime/doc/editing.txt [hunk 2]
```diff
@@ -965,10 +963,10 @@ lost the original file.
 
 						*DOS-format-write*
 If the 'fileformat' is "dos", <CR> <NL> is used for <EOL>.  This is default
-for MS-DOS and Win32. On other systems the message "[dos format]" is shown to
+for Windows. On other systems the message "[dos format]" is shown to
 remind you that an unusual <EOL> was used.
 						*Unix-format-write*
-If the 'fileformat' is "unix", <NL> is used for <EOL>.  On MS-DOS and Win32
+If the 'fileformat' is "unix", <NL> is used for <EOL>.  On Winndows
```
Comment (Pyrohh) [inline runtime/doc/editing.txt]: extra "n" in "Winndows"

File: runtime/doc/insert.txt [hunk 3]
```diff
@@ -1941,9 +1941,9 @@ If the 'fileformats' option is not empty Vim tries to recognize the type of
 changed, the detected format is only used while reading the file.
 A similar thing happens with 'fileencodings'.
 
-On non-MS-DOS and Win32 systems the message "[dos format]" is shown if
+On non-Windows systems the message "[dos format]" is shown if
 a file is read in DOS format, to remind you that something unusual is done.
-On Macintosh, MS-DOS, and Win32 the message "[unix format]" is shown if
+On Macintosh and Windows the message "[unix format]" is shown if
```
Comment (Pyrohh) [inline runtime/doc/insert.txt]: @justinmk I think you mentioned htis before, but were we going to refer to this as `OS X` from now on?

Comment (sethjackson) [inline runtime/doc/insert.txt]: I would prefer so. Yes.

Comment (justinmk) [inline runtime/doc/insert.txt]: I would suggest one word, "OSX". That's how libuv does it in their source code, it makes grepping and web-searching easier, avoid bad word-wrap, etc. Who cares if it's not pedantically correct :P

Comment (sethjackson) [inline runtime/doc/insert.txt]: Heh. I'm pedantic. Whatever we use `OSX` or `OS X` we should just be consistent. I would prefer `OS X` for user docs though.

Comment (Pyrohh) [inline runtime/doc/insert.txt]: > it makes grepping and web-searching easier, avoid bad word-wrap

:+1: but let's do that in another PR

File: runtime/doc/cmdline.txt [hunk 4]
```diff
@@ -919,7 +919,7 @@ option contains "sh", this is done twice, to avoid the shell trying to expand
 the "!".
 
 							*filename-backslash*
-For filesystems that use a backslash as directory separator (MS-DOS and
+For filesystems that use a backslash as directory separator (NTFS on
```
Comment (Pyrohh) [inline runtime/doc/cmdline.txt]: Are there other file systems that windows supports?

Comment (sethjackson) [inline runtime/doc/cmdline.txt]: Well I guess FAT right? Maybe `Windows filesystems`? There are a couple of places to update for this wording BTW.

File: runtime/doc/usr_23.txt [hunk 5]
```diff
@@ -34,7 +34,7 @@ have line-break problems.  The Vim editor automatically recognizes the
 different file formats and handles things properly behind your back.
    The option 'fileformats' contains the various formats that will be tried
 when a new file is edited.  The following command, for example, tells Vim to
-try UNIX format first and MS-DOS format second: >
+try Unix format first and MS-DOS format second: >
```
Comment (Pyrohh) [inline runtime/doc/usr_23.txt]: :+1: since UNIX is the trademarked name (https://archive.org/details/bstj57-6-1905), and I almost never see it anymore.

File: src/nvim/fileio.c [hunk 6]
```diff
@@ -366,14 +366,13 @@ readfile (
   /* don't display the file info for another buffer now */
   need_fileinfo = FALSE;
 
-  /*
-   * For Unix: Use the short file name whenever possible.
-   * Avoids problems with networks and when directory names are changed.
-   * Don't do this for MS-DOS, a "cd" in a sub-shell may have moved us to
```
Comment (Pyrohh) [inline src/nvim/fileio.c]: Are you sure that MS-DOS's behavior here is actually the same as Windows?

File: src/nvim/fileio.c [hunk 7]
```diff
@@ -2353,13 +2352,12 @@ buf_write (
...
-   * another directory, which we don't detect
-   */
-  ffname = fname;                           /* remember full fname */
+
+  // For Unix: Use the short file name whenever possible.
+  // Avoids problems with networks and when directory names are changed.
+  // Don't do this for Windows, a "cd" in a sub-shell may have moved us to
+  // another directory, which we don't detect.
```
Comment (Pyrohh) [inline src/nvim/fileio.c]: same as above

File: src/nvim/os/env.c [hunk 8]
```diff
@@ -142,7 +142,7 @@ void os_get_hostname(char *hostname, size_t len)
 ///   - go to that directory
 ///   - do os_dirname() to get the real name of that directory.
 /// This also works with mounts and links.
-/// Don't do this for MS-DOS, it will change the "current dir" for a drive.
+/// Don't do this for Windows, it will change the "current dir" for a drive.
```
Comment (Pyrohh) [inline src/nvim/os/env.c]: Same here

File: runtime/doc/usr_23.txt (hunk 5 above)
Comment (sethjackson) [inline runtime/doc/usr_23.txt]: Yea there are more refs to `UNIX` IIRC and we shouldn't use that because trademarks. Yay! More doc updates forthcoming. :)

Comment (Pyrohh) [inline runtime/doc/usr_23.txt]: Sounds good, but in another PR, right?

File: src/nvim/fileio.c (hunk 6 above)
Comment (sethjackson) [inline src/nvim/fileio.c]: I would assume so yes but I did not look into it. AFAIK cd will change the current directory.

However if you look at the Vim source `BACKSLASH_IN_FILENAME` is defined in `os_dos.h` which gets included in `os_win32.h` so I would assume we want this same behavior.

Just FYI we do not define `BACKSLASH_IN_FILENAME` yet for Windows but we should AFAIK.

File: runtime/doc/usr_23.txt (hunk 5 above)
Comment (sethjackson) [inline runtime/doc/usr_23.txt]: Yes. :)

File: runtime/doc/options.txt [hunk 9]
```diff
@@ -5287,7 +5287,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 
 						*'shell'* *'sh'* *E91*
 'shell' 'sh'		string	(default $SHELL or "sh",
-					MS-DOS and Win32: "command.com" or
+					Windows: "command.com" or
```
Comment (Pyrohh) [inline runtime/doc/options.txt]: I don't think this is true; from `options.lua`:
```
    {
      full_name='shell', abbreviation='sh',
      type='string', scope={'global'},
      secure=true,
      vi_def=true,
      expand=true,
      varname='p_sh',
      defaults={
        condition='WIN32',
        if_true={vi="cmd.exe"},
        if_false={vi="sh"}
      }
    },
```

Comment (Pyrohh) [inline runtime/doc/options.txt]: There are a few other references of `command.com` in this file by the way.

File: runtime/doc/options.txt [hunk 10]
```diff
@@ -5330,11 +5330,11 @@ A jump table for the options with a short description can be found at |Q_op|.
 
 						*'shellcmdflag'* *'shcf'*
 'shellcmdflag' 'shcf'	string	(default: "-c";
-				 MS-DOS and Win32, when 'shell' does not
+				 Windows, when 'shell' does not
 				 contain "sh" somewhere: "/c")
 			global
 	Flag passed to the shell to execute "!" and ":!" commands; e.g.,
-	"bash.exe -c ls" or "command.com /c dir".  For the MS-DOS-like
+	"bash.exe -c ls" or "command.com /c dir".  For Windows
```
Comment (Pyrohh) [inline runtime/doc/options.txt]: same regarding command.com

File: src/nvim/fileio.c (hunk 6 above)
Comment (sethjackson) [inline src/nvim/fileio.c]: See #465 too.

**Comment by Pyrohh** [comment]
Just so I don't forget to mention it: please don't amending your commits or rebase, as then I'd have to re-review everything. New commits are preferable.

File: runtime/doc/options.txt (hunk 9 above)
Comment (sethjackson) [inline runtime/doc/options.txt]: Yes there are lies about `command.com` and `%COMSPEC%` too.

But we can safely change `command.com` to `cmd.exe` I think in the docs.

**Comment by sethjackson** [comment]
Ok I can add new commits instead of `git commit --amend`.

File: runtime/doc/usr_23.txt (hunk 5 above)
Comment (Pyrohh) [inline runtime/doc/usr_23.txt]: It's fine, that would be for little gain IMO.

File: src/nvim/fileio.c (hunk 6 above)
Comment (Pyrohh) [inline src/nvim/fileio.c]: I found http://superuser.com/a/316515, too, so I guess you're right.

File: runtime/doc/options.txt [hunk 11]
```diff
@@ -6342,9 +6342,8 @@ A jump table for the options with a short description can be found at |Q_op|.
 'term'			string	(default is $TERM, if that fails:
 				      in the GUI: "builtin_gui"
 					  on Mac: "mac-ansi"
-				       on MS-DOS: "pcterm"
```
Comment (Pyrohh) [inline runtime/doc/options.txt]: There are some more refs to pcterm:
```diff
diff --git a/runtime/doc/options.txt b/runtime/doc/options.txt
index 1a9c951..79db9b1 100644
--- a/runtime/doc/options.txt
+++ b/runtime/doc/options.txt
@@ -779,7 +779,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 
 	Normally this option would be set in the vimrc file.  Possibly
 	depending on the terminal name.  Example: >
-		:if &term == "pcterm"
+		:if &term == "xterm"
 		:  set background=dark
 		:endif
 <	When this option is set, the default settings for the highlight groups
@@ -7099,6 +7099,6 @@ A jump table for the options with a short description can be found at |Q_op|.
 			global
 	The number of microseconds to wait for each character sent to the
 	screen.  When non-zero, characters are sent to the terminal one by
-	one.  For MS-DOS pcterm this does not work.  For debugging purposes.
+	one.  For debugging purposes.
 
  vim:tw=78:ts=8:ft=help:noet:norl:
diff --git a/runtime/doc/term.txt b/runtime/doc/term.txt
index 961ec1b..9c74a7e 100644
--- a/runtime/doc/term.txt
+++ b/runtime/doc/term.txt
@@ -425,11 +425,9 @@ cleared when scrolling).
 Unfortunately it is not possible to deduce from the termcap how cursor
 positioning should be done when using a scrolling region: Relative to the
 beginning of the screen or relative to the beginning of the scrolling region.
-Most terminals use the first method.  A known exception is the MS-DOS console
-(pcterm).  The 't_CS' option should be set to any string when cursor
-positioning is relative to the start of the scrolling region.  It should be
-set to an empty string otherwise.  It defaults to "yes" when 'term' is
-"pcterm".
+Most terminals use the first method.  The 't_CS' option should be set to any
+string when cursor positioning is relative to the start of the scrolling
+region.  It should be set to an empty string otherwise.
 
 Note for xterm users: The shifted cursor keys normally don't work.  You can
 	make them work with the xmodmap command and some mappings in Vim.
```

File: runtime/doc/pattern.txt [hunk 12]
```diff
@@ -131,7 +131,7 @@ gD			Goto global Declaration.  When the cursor is on a
 
 							*CTRL-C*
 CTRL-C			Interrupt current (search) command.  Use CTRL-Break on
-			MS-DOS |dos-CTRL-Break|.
+			Windows |dos-CTRL-Break|.
```
Comment (sethjackson) [inline runtime/doc/pattern.txt]: I'm not too sure about this one too. I'm fairly certain you can interrupt with `CTRL-C` on Windows...

File: runtime/doc/usr_23.txt (hunk 5 above)
Comment (sethjackson) [inline runtime/doc/usr_23.txt]: Ok.

File: runtime/doc/usr_27.txt [hunk 13]
```diff
@@ -83,7 +83,7 @@ matter if 'ignorecase' or 'smartcase' was changed.
 
 	Note:
 	If your search takes much longer than you expected, you can interrupt
-	it with CTRL-C on Unix and  CTRL-Break on MS-DOS and MS-Windows.
+	it with CTRL-C on Unix and CTRL-Break on Windows.
```
Comment (sethjackson) [inline runtime/doc/usr_27.txt]: Same thing about `CTRL-Break`. Not sure. Although this would seem to indicate it works both on DOS and Windows... Still think `CTRL-C` should work though...

File: src/nvim/fileio.c (hunk 6 above)
Comment (sethjackson) [inline src/nvim/fileio.c]: :+1: we still need `BACKSLASH_IN_FILENAME` defined somewhere though I think. I couldn't find it in `master` or in a commit in #810.

But that's for another PR.

File: runtime/doc/usr_27.txt (hunk 13 above)
Comment (Pyrohh) [inline runtime/doc/usr_27.txt]: [This page](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682541%28v=vs.85%29.aspx) seems to agree with you. Do you have a system with Windows that this can be tested on? If Neovim doesn't work on Windows yet, then I bet Vim could still be used to demonstrate `CTRL-C` behavior in console.

File: runtime/doc/syntax.txt [hunk 14]
```diff
@@ -4507,7 +4507,7 @@ mentioned for the default values. See |:verbose-cmd| for more information.
 					*highlight-args* *E416* *E417* *E423*
 There are three types of terminals for highlighting:
 term	a normal terminal (vt100, xterm)
-cterm	a color terminal (MS-DOS console, color-xterm, these have the "Co"
+cterm	a color terminal (Wondows console, color-xterm, these have the "Co"
```
Comment (Pyrohh) [inline runtime/doc/syntax.txt]: Microsoft Wondows™

File: runtime/doc/usr_27.txt (hunk 13 above)
Comment (sethjackson) [inline runtime/doc/usr_27.txt]: I'll try with Vim. Neovim doesn't work in the terminal on Windows so can't test that currently.

What's a good way to test this?

```
:sleep <value>
<CTRL-C>
```

I'll admit my ignorance about never using the sleep feature. :)

Above seems to work on Neovim on Unix though...

File: runtime/doc/syntax.txt (hunk 14 above)
Comment (sethjackson) [inline runtime/doc/syntax.txt]: It's Wondowsful!

File: runtime/doc/usr_27.txt (hunk 13 above)
Comment (Pyrohh) [inline runtime/doc/usr_27.txt]: :+1: That seems like it would work.

File: runtime/doc/term.txt [hunk 15]
```diff
@@ -689,9 +688,9 @@ work on systems where the window manager consumes the mouse events when the
 alt key is pressed (it may move the window).
 
 							*double-click*
-Double, triple and quadruple clicks are supported when the GUI is active,
-for MS-DOS and Win32, and for an xterm (if the gettimeofday() function is
-available).  For selecting text, extra clicks extend the selection:
+Double, triple and quadruple clicks are supported when the GUI is active, for
+Windows and for an xterm (if the gettimeofday() function is available).  For
```
Comment (Pyrohh) [inline runtime/doc/term.txt]: I think the `gettimeofday()` note can be deleted; see #975.

File: runtime/doc/usr_02.txt [hunk 16]
```diff
@@ -29,8 +29,8 @@ To start Vim, enter this command: >
 
 	gvim file.txt
 
-In UNIX you can type this at any command prompt.  If you are running Microsoft
-Windows, open an MS-DOS prompt window and enter the command.
+On Unix you can type this at any command prompt.  If you are running Microsoft
+Windows, open an command prompt window and enter the command.
```
Comment (Pyrohh) [inline runtime/doc/usr_02.txt]: `an command` -> `a command`

File: runtime/doc/usr_27.txt (hunk 13 above)
Comment (sethjackson) [inline runtime/doc/usr_27.txt]: `CTRL-C` doesn't work on Console Vim on Windows it appears. It appears to work in Gvim though.

**Comment by Pyrohh** [comment]
Okay, done reviewing.

File: runtime/doc/usr_02.txt (hunk 16 above)
Comment (sethjackson) [inline runtime/doc/usr_02.txt]: Fixed thanks.

File: runtime/doc/term.txt (hunk 15 above)
Comment (sethjackson) [inline runtime/doc/term.txt]: :+1: removed.

File: runtime/doc/usr_27.txt (hunk 13 above)
Comment (Pyrohh) [inline runtime/doc/usr_27.txt]: Thanks for looking into it, I guess the notice should be kept.

File: runtime/doc/options.txt (hunk 11 above)
Comment (sethjackson) [inline runtime/doc/options.txt]: :+1: fixed thanks.

**Comment by sethjackson** [comment]
I believe I have handled all the comments. Lint is failing because it seems to get confused about `forward_slash` vs `for`. :(

**Comment by Pyrohh** [comment]
> I believe I have handled all the comments. Lint is failing because it seems to get confused about forward_slash vs for. :(

@ZyX-I might want to know about this.

**Comment by sethjackson** [comment]
Yea I don't think it's because I messed something up but maybe I did. Very strange.

**Comment by Pyrohh** [comment]
Since the linter errors seem bogus, I just commited this. AFAICT the warnings should be added to the linter's suppression list.

Anway, thanks a lot @sethjackson.
